core/blockchain.go

```diff
@@ -1207,10 +1207,22 @@ func (self *BlockChain) update() {
 }
 
 // reportBlock logs a bad block error.
-func reportBlock(block *types.Block, err error) {
+func (bc *BlockChain) reportBlock(block *types.Block, receipts types.Receipts, err error) {
 	if glog.V(logger.Error) {
-		glog.Errorf("Bad block #%v (%s)\n", block.Number(), block.Hash().Hex())
-		glog.Errorf("    %v", err)
+		var receiptString string
+		for _, receipt := range receipts {
+			receiptString += fmt.Sprintf("\t%v\n", receipt)
+		}
+		glog.Errorf(`########## BAD BLOCK #########
+Chain config: %v
+Number: %v
+Hash: 0x%x
+%v
+
+Error: %v
+`, bc.config, block.Number(), block.Hash(), receiptString, err)
+		glog.V(logger.Detail).Infoln(block)
+		glog.Error("###############################")
```

Comment:
The log looks a bit strange this way that the hashes are in the same line as the log entry but the text not:

```
E1123 13:54:58.229591 core/blockchain.go:1223] ########## BAD BLOCK #########
Chain config: {ChainID: 2 Homestead: 494000 DAO: <nil> DAOSupport: true EIP150: 1885000 EIP155: <nil> EIP158: 1885000}
Number: 1
Hash: 0x41800b5c3f1717687d85fc9018faac0a6e90b39deaa0b99e7fe4fe796ddeb26a


Error: Difficulty check failed for header (remote: 997888 local: 1048064)
E1123 13:54:58.229629 core/blockchain.go:1225] ###############################
```

Wouldn't it make more sense to add a newline before the "header" and leave the "footer" in the same block too and not use a new log entry?

Also the header is 1 hash shorter than the footer.